config.go

```diff
@@ -331,6 +331,24 @@ func (b *Builder) SetUser(spec string) {
 	b.Docker.Config.User = spec
 }
 
+// OnBuild returns the OnBuild value from the container.
+func (b *Builder) OnBuild() []string {
+	return copyStringSlice(b.Docker.Config.OnBuild)
+}
+
+// ClearOnBuild removes all values from the OnBuild structure
```

Comment:
Is this strictly necessary?  I wasn't running into an issue with left over cruft in the OnBuild.

Reply:
FWIW, I tested without this function being called and it worked fine.  I don't know if you'd other scenarios in mind that this would handle.